custom_components/alexa_media/alexa_entity.py

```diff
@@ -188,6 +204,33 @@ def parse_alexa_entities(network_details: Optional[Dict[Text, Any]]) -> AlexaEnt
                         serial if serial else appliance["entityId"]
                     )
                     temperature_sensors.append(processed_appliance)
+                # Code for Amazon Smart Air Quality Monitor
+                elif is_air_quality_sensor(appliance):
+                    serial = get_device_serial(appliance)                    
+                    processed_appliance["device_serial"] = (
+                        serial if serial else appliance["entityId"]
+                    )                    
+                    # create array of air quality sensors. We must store the instance id against
+                    # the assetId so we know which sensors are which.
+                    sensors = []
+                    if appliance["friendlyDescription"] == "Amazon Indoor Air Quality Monitor" :
+                        for cap in appliance["capabilities"]:
+                            instance = cap.get("instance")
+                            if instance :                                
+                                friendlyName = cap["resources"].get("friendlyNames")
```

Comment:
Is it pretty clear the key `resources` will exist in `instance` exists? Want to avoid keyerrors for the majority of users who do not have IAQM.

Reply:
This code should only run if the appliance matches based on below.
if appliance["friendlyDescription"] == "Amazon Indoor Air Quality Monitor" :
I can only test against my device, but yes if instance existed there was always resources too.

Reply:
Is there any neater way of traversing down the keys than using get? Is get the only safe way if you are unsure if the key is there?

Reply:
I had missed it was nested in that if. That should be fine. But safest way is either nested gets or checking for keys before access.

Each level of nesting can be done like this with a get.
```python
something.get("key1", {}).get("key2", DEFAULT)
```